store/NodeRedisStore.js

```
@@ -0,0 +1,52 @@
+'use strict'
+
+/**
+ * When using node-redis, you need to initialize the client with the rateLimit script like this:
+ * ```js
+ * const redis = createClient({
+ *   scripts: {
+ *     rateLimit: rateLimit.NodeRedisStore.rateLimitScript
+ *   }
+ * });
+ * ```
+ */
+
+const { lua } = require('./RedisStore')
+const { defineScript } = require('@redis/client')
+
+const rateLimitScript = defineScript({
+  NUMBER_OF_KEYS: 1,
+  SCRIPT: lua,
+  transformArguments (key, timeWindow, max, continueExceeding, exponentialBackoff) {
+    return [key, String(timeWindow), String(max), String(continueExceeding), String(exponentialBackoff)]
+  },
+  transformReply (reply) {
+    return reply
+  },
+})
+
+function NodeRedisStore (continueExceeding, exponentialBackoff, redis, key = 'fastify-rate-limit-') {
+  this.continueExceeding = continueExceeding
+  this.exponentialBackoff = exponentialBackoff
+  this.redis = redis
+  this.key = key
```

Should throw an Error here with hint how to `createClient` - passing the script?